applications/MPMApplication/custom_strategies/schemes/mpm_residual_based_bossak_scheme.hpp

```diff
@@ -387,6 +413,35 @@ class MPMResidualBasedBossakScheme
         KRATOS_CATCH( "" )
     }
 
+    /**
+     * @brief Function called once at the end of a solution step, after convergence is reached if an iterative process is needed
+     * @param rModelPart The model part of the problem to solve
+     * @param A LHS matrix
+     * @param Dx Incremental update of primary variables
+     * @param b RHS Vector
+     */
+    void FinalizeSolutionStep(
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb) override
+    {
+        BossakBaseType::FinalizeSolutionStep(rModelPart, rA, rDx, rb);
+
+
+        #pragma omp parallel for
```

Comment:
Ah okay thanks for letting me know, I'll make the neccessary changes - I wasn't aware since the existing loops in the application were written this way.